### DISCUSSION TITLE
How to use pyright with eglot the right way

### DISCUSSION THREAD

**Discussion by buhtz** [joaotavora/eglot#939] [General]
I switched from `python-language-server` to `pyright` hoping this is better for the ressources of my small RaspberryPi4.

For me it is quit unclear how Emacs, eglot, pyright and maybe some other tools (e.g. flymake, pep8, pycodestyle, ...) are working together and which one is responsible for what one. ;)

It seems that `eglot` works out of the box with `pyright`. I have no problems here.

But I wonder what possible configurations there are. Do I need or should install some other tools (e.g. pycodestyle, etc) in the background to improve the setup? Where can I read about that; should I look at eglot or pyright for that?

I also see that the minibuffer now shows me nice formated and colored infos about the function signatures. With pyls in the past it wasn't that beautiful. ;)

**Comment by doolio** [comment]
I believe such features depend on the server and I don't believe `pyright` provides those features.

**Comment by jdtsmith** [comment]
One thing I figured out is that eglot replaces the flymake functions with its own, assuming the server will handle all the linting.  pyright doesn't do so, but you can add flymake linking capability back in; see [this comment](https://www.reddit.com/r/emacs/comments/s96wec/comment/ii6norq/?utm_source=share&utm_medium=web2x&context=3). 

BTW, can you post a picture of your "nicely colored signatures"?  I get really bad/ugly signatures with pyright, e.g. for `range()`:

![image](https://user-images.githubusercontent.com/93749/190872030-9a62e0bb-984c-4590-9a14-cca82e736a51.png)
